src/detect-flowbits.c

```diff
@@ -91,6 +91,26 @@ void DetectFlowbitsRegister (void)
     DetectSetupParseRegexes(PARSE_REGEX, &parse_regex);
 }
 
+static void DetectFlowbitValidateCallback(Signature *s, bool postmatch, DetectFlowbitsData *fd)
+{
+    SigMatch *list = NULL;
+    if (!postmatch && fd->cmd == DETECT_FLOWBITS_CMD_ISSET)
+        list = s->init_data->smlists[DETECT_SM_LIST_POSTMATCH];
+    else if (postmatch && fd->cmd == DETECT_FLOWBITS_CMD_SET)
+        list = s->init_data->smlists[DETECT_SM_LIST_MATCH];
+    for (const SigMatch *sm = list; sm != NULL; sm = sm->next) {
+        if (sm->type != DETECT_FLOWBITS)
+            continue;
+        /* Check if an entry corresponding to the same idx exists in the opposite list */
+        DetectFlowbitsData *fd2 = (DetectFlowbitsData *)sm->ctx;
+        if (fd2->idx == fd->idx) {
```

Comment:
should we have a more explicit check about `fd2->cmd` ?

Reply:
indeed that'll be better. this leaves scope for bugs